corehq/apps/api/resources/messaging_event/utils.py

```diff
@@ -74,10 +74,11 @@ def get_request_params(request):
     """Get the query params from the request. If the request is using a cursor
     decode the cursor and return those values as the query params.
     :returns: CursorParams tuple"""
+    has_cursor = False
     request_params = request.GET.dict()
     if 'cursor' in request_params:
-        params_string = b64decode(request_params['cursor']).decode('utf-8')
-        return CursorParams(
-            QueryDict(params_string).dict(), request.domain, True
-        )
-    return CursorParams(request_params, request.domain, False)
+        has_cursor = True
+        cursor_params_dict = QueryDict(b64decode(request_params['cursor']).decode('utf-8')).dict()
```

Comment:
I wasn't sure whether to add a comment here, but I will with a caveat: What you've written here is fine. You don't need to change anything.

My comment was going to be that lines where several things are happening, like this one, take me a couple of moments to parse. I try to write code as if I'm trying to explain what I'm doing to my future self, or to someone else. So when I think about it that way, firstly I tend to write shorter lines, and secondly the process of trying to explain what I'm doing often shows me ways of simplifying it.

If I was trying to explain this code to my future self, I might have written it like this:
```python
def get_request_params(request):
    has_cursor = False
    request_params = request.GET.dict()
    if 'cursor' in request_params:
        has_cursor = True
        b64_cursor = request_params.pop('cursor')
        cursor_str = b64decode(b64_cursor).decode('utf-8')
        cursor_params_dict = QueryDict(cursor_str).dict()
        request_params.update(cursor_params_dict)
    return CursorParams(request_params, request.domain, has_cursor)
```

Reply:
Thanks for highlighting. I'll do this refactor in a followup PR :+1: